lib/src/main/java/com/squareup/cycler/Update.kt

```diff
@@ -67,6 +104,7 @@ class Update<I : Any>(private val oldRecyclerData: RecyclerData<I>) {
   var detectMoves: Boolean = true
 
   private val addedChunks = mutableListOf<List<I>>()
+  private val dataPopulated get() = oldRecyclerData.data.isEmpty && !data.isEmpty
```

Comment:
I'd do:
dataPopulated get() = oldRecyclerData.data.isEmpty // if new data is empty as well, it's populated too
dataReplaced get() = !dataPopulated && old.data != data
dataAdded get() = !dataReplaced && !dataPopulated && addedChunks.isNotEmpty()